django/db/backends/sqlite3/features.py

```diff
@@ -91,10 +90,6 @@ def django_test_skips(self):
 
     @cached_property
     def supports_atomic_references_rename(self):
-        # SQLite 3.28.0 bundled with MacOS 10.15 does not support renaming
-        # references atomically.
-        if platform.mac_ver()[0].startswith('10.15.') and Database.sqlite_version_info == (3, 28, 0):
```

Comment:
Yes... our hope this was just the one version proved wrong.

Reply:
Ah whilst reading the issues back, I hadn't noticed this used such a narrow macOS version. Perhaps we should gate the `PRAGMA` query under this platform check, to avoid an unnecessary query on every connection.

Reply:
Hey @adamchainz — sorry I missed this comment before merging. 

As per https://github.com/django/django/pull/15158#pullrequestreview-825140230 I think we need this always for SQLite on macOS — it's not version specific AFAICS. 

If you think there's a performance worry with the `PRAGMA` (really? 🤔) — then please do follow-up!

Reply:
> If you think there's a performance worry with the PRAGMA (really? 🤔) — then please do follow-up!

Yeah, it's not a huge worry, because it's SQLite. For client/server DB's we'd want to avoid the extra roundtrip.